### ISSUE TITLE
macOS: Cannot select text with Shift+Arrow in tree view, rotates view instead

### ISSUE THREAD

**Issue by maxwxyz** [FreeCAD/FreeCAD#14869]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

From: https://forum.freecad.org/viewtopic.php?p=767281
Trying to select characters with Shift + Left or Right Arrow key on macOS in the tree view does not select the text but rotates the view. I cannot reproduce this on Windows.

Steps to reproduce:
1. create a new object
2. right click "Rename"
3. try to select with keyboard shurtcuts (Shift + Left Arrow)

<img width="309" alt="grafik" src="https://github.com/FreeCAD/FreeCAD/assets/6246609/903716b6-3028-412f-9408-936330c68452">

Other reported the same issue in the file save dialog but I cannot reproduce this.

### Full version info

```shell
OS: macOS 13.6.7
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37841 (Git)
Build type: Release
Branch: main
Hash: 5f761c758e2a0164b00a7183b9c0cbe32688e123
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Installed mods: 
  * 3DfindIT 1.2.0
  * freecad.gears 1.2.0
  * OpenTheme 2024.6.22
  * CfdOF 1.25.17
  * Defeaturing 1.2.2
  * Manipulator 1.5.7
  * fasteners 0.5.24
  * CurvedShapes 1.0.9
  * sheetmetal 0.4.19
  * Curves 0.6.39
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
@maxwxyz is there anyone working on macos related tickets ?

**Comment by maxwxyz** [comment]
@luzpaz not sure

**Comment by luzpaz** [comment]
@JohnOCFII would you mind reproducing ?

**Comment by JohnOCFII** [comment]
> @JohnOCFII would you mind reproducing ?

Yes, I can reproduce this.  Standard macOS behavior when editing a text field and holding SHIFT and arrow left or right should continuously select text.  For some reason, even though I am editing (in this case, attempting to rename a sketch) the keypress is rotating the object in PartDesign.

```
OS: macOS 14.5
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38261 (Git)
Build type: Release
Branch: main
Hash: 51789ae8a1a0b1ed20317eaedcd190baac734e3d
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Installed mods: 
  * FeedsAndSpeeds 0.5.0
  * OpenTheme 2024.7.3
```
<img width="1621" alt="Screenshot 2024-07-21 at 2 46 16 PM" src="https://github.com/user-attachments/assets/c99171fd-98b4-4abb-85c6-2984fee97bfe">

**Comment by JohnOCFII** [comment]
Also confirmed same issue with the File export Dialog.  (same FreeCAD version as earlier report today 38261)
<img width="1511" alt="Screenshot 2024-07-21 at 8 12 43 PM" src="https://github.com/user-attachments/assets/b4cd19b8-d44e-4297-8b66-490506098ac7">

**Comment by ryanthara** [comment]
I can confirm this behavior in the file dialog with version 

`OS: macOS 15.2
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: C/Default (C)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
`

**Comment by luzpaz** [comment]
Anyone interested in spearheading this ?

**Comment by abluescarab** [comment]
This is happening to me on Windows as well.

**Comment by timpieces** [comment]
I wonder if this is related to a similar issue that I have where when I'm (e.g.) saving a document, and I press 'Cmd+a' to select the document name (in the new save dialog), it selects *everything in freecad*.
This still reproduces for me, so I'll take a look at it, as that bug annoys me quite a lot.

**Comment by timpieces** [comment]
Ok so from my investigation:
- The issue I was mentioning with Cmd+a and this issue do indeed seem to be the same thing.
- Online documentation for this issue seems scarce, but it seems that MacOS has slightly different behaviour for application-level hotkeys. The TLDR seems to be that if you register a shortcut for something that you put in the application menu bar in QT, then that shortcut will override anything. This doesn't seem to be true of other platforms.

I've looked for a lot of workarounds and there's not many good ones. It seems like there are two potential approaches:
- address every possible widget (e.g. the widget to edit object label) and make it specifically override these menu-bar level shortcuts
- split these global shortcuts in two such that:
    - they are added to the menu bar with no shortcut, and
    - they are added to a locally scoped widget with their shortcut
The second of these approaches would have the effect that the shortcut would no longer work on other platforms when the window isn't focused (e.g. if you have clicked on the tree view then shift+right would no longer rotate the 3d view).

I will probably try to look a little bit further, but it may be hard to fix this functionality on mac without having impacts on other platforms, and it may not be as straightforward of a fix as I'd have hoped.
I was surprised that more applications don't have these issues, and I'd like to look into that further, but I think it's rare to add so many menu bar shortcuts with these types of shortcuts

**Comment by timpieces** [comment]
Looked further. Initially ShortcutOverride (https://wiki.qt.io/ShortcutOverride) wasn't working for me, but it turns out I was matching the wrong key-combination.
I think there is a sensible solution that leverages ShortcutOverride, and I have a proof-of-concept that works. I'll look to turn it into a PR.

While the Cmd+A in save dialog issue is a similar root cause, the same fix doesn't work. I'm investigating other ways to fix this, but probably it will be more complex and I'll create a separate issue for it if it doesn't exist already
